doozerlib/runtime.py

```diff
@@ -818,6 +835,37 @@ def rpm_metas(self) -> List[RPMMetadata]:
     def all_metas(self) -> List[Union[ImageMetadata, RPMMetadata]]:
         return self.image_metas() + self.rpm_metas()
 
+    def get_payload_image_metas(self) -> List[ImageMetadata]:
+        """
+        :return: Returns a list of ImageMetadata that are destined for the OCP release payload. Payload images must
+                    follow the correct naming convention or an exception will be thrown.
+        """
+        payload_images = []
+        for image_meta in self.image_metas():
+            if image_meta.is_payload:
```

Comment:
I think it should filter base-only images first like what we previously do in gen_payload:

```suggestion
            if not image_meta.base_only and image_meta.is_payload:
```

Reply:
@vfreex / @sosiouxme - can we just add `not image_meta.base_only` to `image_meta.is_payload`? Or just raise an exception in is_payload if the image claims both base_only and payload?
Seems strange to need to check both, but I may be missing something.

Reply:
Personally +1 on moving the check to image_meta.is_payload.

Reply:
ack. Adding it as an error in is_payload so we fail fast.